sdk/identity/azure-identity-v2/src/main/java/com/azure/identity/v2/AzurePowerShellCredential.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.v2;
+
+import com.azure.identity.v2.implementation.client.DevToolsClient;
+import com.azure.identity.v2.implementation.models.DevToolsClientOptions;
+import com.azure.identity.v2.implementation.util.LoggingUtil;
+import com.azure.v2.core.credentials.TokenCredential;
+import com.azure.v2.core.credentials.TokenRequestContext;
+import io.clientcore.core.credentials.oauth.AccessToken;
+import io.clientcore.core.instrumentation.logging.ClientLogger;
+
+/**
+ * <p>The Azure Powershell is a command-line tool that allows users to manage Azure resources from their local machine
+ * or terminal. It allows users to
+ * <a href="https://learn.microsoft.com/powershell/azure/authenticate-azureps">authenticate interactively</a>
+ * as a user and/or a service principal against
```

Comment:
same "should we push service principal" question here.

Reply:
we will do a formal pass later before GA too.